src/coreclr/vm/callstubgenerator.cpp

```diff
@@ -2416,7 +2599,10 @@ void CallStubGenerator::TerminateCurrentRoutineIfNotOfNewType(RoutineType type,
         m_r1 = NoRange;
         m_currentRoutineType = RoutineType::None;
     }
-    else if ((m_currentRoutineType == RoutineType::FPReg) && (type != RoutineType::FPReg))
+#ifndef TARGET_ARM
+    else
```

Comment:
I don't understand why this ifdef is needed.

Reply:
Also, the case when a struct is split between registers and stack also exists on x64 Linux. The ArgIterator has the GetArgLocDescForStructInRegs() for this purpose. The code in the ComputeCallStub uses that to pretend the parts of the struct are "separate arguments" from the further processing point of view. The stuff around it is quite complex due to the complicated classification mechanism defined by the Unix amd64 ABI, but it seems we could handle the case of args split between regs and stack at that place too and I think we would not have to handle that here then.